### PR TITLE
Rewrite metasync.filter without sorting

### LINKED ISSUE #55: Avoid sort in metasync.filter
Now it use push and sort. It is better to allocate `new Array(len)` and remove filtered. But tests needed.

### PR DESCRIPTION
**Pull request by nechaido**
Fixes #55

### CODE REVIEW

**Comment by nechaido** [comment]
@tshemsedinov ping

**Comment by tshemsedinov** [comment]
@nechaido also rebased

File: lib/array.js [hunk 1]
```diff
@@ -45,43 +45,34 @@ const filter = (
...
+  items.forEach((item) => {
+    fn(item, (err, accepted) => {
+      if (finished) return;
+      if (err) {
+        finished = true;
+        done(err);
+        return;
+      }
+      if (accepted) result[count++] = item;
+      else len--;
```
Comment (nechaido) [inline lib/array.js]: @tshemsedinov I'd prefer this to be:
```
if (accepted) {
  result[count++] = item;
} else {
  len--;
}
```
But it is up to you.

Comment (tshemsedinov) [inline lib/array.js]: I prefer short syntax for single operator block, you know, @nechaido

**Comment by tshemsedinov** [comment]
Landed in ddd71bb

**Pull request closed without merging**